src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react'
+import { createPortal } from 'react-dom'
+
+const uuid = () => {
```

Comment:
Not sure why you're generating the uuid like this I prefer you install the uuid library to do this.